runtests.py

```diff
@@ -328,19 +328,21 @@ def build_project(args):
     # Always use ccache, if installed
     env['PATH'] = os.pathsep.join(EXTRA_PATH + env.get('PATH', '').split(os.pathsep))
     cvars = distutils.sysconfig.get_config_vars()
-    if 'gcc' in cvars.get('CC', ''):
-        # add flags used as werrors
-        warnings_as_errors = ' '.join([
-            # from tools/travis-test.sh
-            '-Werror=declaration-after-statement',
-            '-Werror=vla',
-            '-Werror=nonnull',
-            '-Werror=pointer-arith',
-            '-Wlogical-op',
-            # from sysconfig
-            '-Werror=unused-function',
-        ])
-        env['CFLAGS'] = warnings_as_errors + ' ' + env.get('CFLAGS', '')
+    if 'gnu' in cvars.get('CC', ''):
```

Comment:
Perhaps we are over-thinking this. I added these warning flags so a run of `runtests.py` would fail in the same way as a CI build. However the CI build has slightly different logic - it caches the output of the `build_ext` step and greps for warnings, filtering out those that are part of the configuration phase. 

So IMO this `if` clause could be overly-specific without any real harm, the "real" test is the CI one which checks for **any** compiler warning. Building here without the flags means the warnings will only be caught by the CI system, inconveniencing the person trying to submit a pull request.

Another option is to cache and parse the `build_ext` output in `runtest.py` like is done for the linux build on CI, but that was more complicated for me to figure out how to do without breaking non-linux builds.

Anyone who specifies a non-gnu compiler will have to find out how to add the flags themselves or wait for the CI failure.

Reply:
> Perhaps we are over-thinking this.

I don't think so, just changing `gnu` to `gcc` and it's good to go. The old code was fine too, except it didn't anticipate the "clang pretending to be gcc" on macOS.

> Another option is to cache and parse the build_ext output in runtest.py 

that would be odd imho - compiler flags are meant for this, so using them is fine

Reply:
> Still not 100% correct I think, this should be `'gcc'`. Otherwise `CC=gcc` will still not get these flags added.

Ah, you are right. Fixed and force-pushed.